domains/certificates/Shared/DomainCertificate/ValueObjects/Gsrn.cs

```diff
@@ -0,0 +1,19 @@
+namespace DomainCertificate.ValueObjects;
+
+public class Gsrn : ValueObject
+{
+    public string Value { get; }
+
+    public Gsrn(string value)
+    {
+        if (value.Length != 18 || !value.StartsWith("57"))
+            throw new ArgumentException("GSRN must be 18 characters long and start with 57.");
```

Comment:
It must be 18 digits. Regex: `^57\d{16}$`

Reply:
Regex expert :) corrected